src/server/services/saml.strategy.ts

```diff
@@ -0,0 +1,47 @@
+import { Strategy } from 'passport-saml';
+import { PassportStrategy } from '@nestjs/passport';
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { UserData, HarvardKeyProfile } from '../interfaces';
+import { ConfigService } from './config.service';
+
+/**
+ * Implements passport-saml to connect to Harvard Key for authentication
+ */
+
+@Injectable()
+class SAMLStrategy extends PassportStrategy(Strategy) {
+  private readonly devMode: boolean;
+
+  public constructor(config: ConfigService) {
+    super({
+      entryPoint: config.get('CAS_URL'),
+      issuer: 'passport-saml',
+      host: config.get('EXTERNAL_URL'),
+    });
+    this.devMode = !config.isProduction;
+  }
+
+  public async validate(profile: HarvardKeyProfile): Promise<UserData> {
+    if (this.devMode) {
+      return {
```

Comment:
I would recommend turning `UserData` into a class (essentially a DTO) and then returning an instance of that instead. This way, we don't have different data formats floating around (hopefully)

Reply:
Added the user class in ac76a66

Is that what you had in mind?